src/chart/generator/plot.cpp

```diff
@@ -102,29 +107,109 @@ bool Plot::dimensionMatch(const Plot &a, const Plot &b)
 }
 bool Plot::hasMarkerChange(const Plot &source, const Plot &target)
 {
-	return source.markers.size() != target.markers.size()
-	    || source.markersInfo != target.markersInfo;
+	auto msize = source.markers.size();
+	if (msize != target.markers.size()
+	    || source.markersInfo != target.markersInfo)
+		return true;
+
+	for (std::size_t ix = 0; ix < msize; ++ix)
+		if (source.markers[ix].idx != target.markers[ix].idx)
+			return true;
+	return false;
 }
 void Plot::mergeMarkersAndCellInfo(Plot &source, Plot &target)
 {
 	auto &&smarkers = source.markers;
 	auto &&tmarkers = target.markers;
-	auto &&ssize = smarkers.size();
-	auto &&tsize = tmarkers.size();
-	for (auto i = ssize; i < tsize; ++i)
-		smarkers.emplace_back(tmarkers[i]).enabled = false;
+	auto first1 = smarkers.begin();
+	auto first2 = tmarkers.begin();
+	std::vector<Marker::MarkerIndexPosition> source_reindex(
+	    smarkers.size());
+	std::vector<Marker::MarkerIndexPosition> target_reindex(
+	    tmarkers.size());
+	auto reindex1 = source_reindex.begin();
+	auto reindex2 = target_reindex.begin();
+	std::size_t pos{};
+	while (first1 != smarkers.end() && first2 != tmarkers.end()) {
+		if (auto cmp = first1->idx <=> first2->idx; std::is_lt(cmp)) {
+			*reindex1++ = {first1->idx, pos};
+			first2 = tmarkers.emplace(first2, *first1);
+			first2->enabled = false;
+		}
+		else if (std::is_gt(cmp)) {
+			*reindex2++ = {first2->idx, pos};
+			first1 = smarkers.emplace(first1, *first2);
+			first1->enabled = false;
+		}
+		else {
+			*reindex1++ = *reindex2++ = {first1->idx, pos};
+		}
+		++first1;
+		++first2;
+		++pos;
+	}
+
+	if (first2 != tmarkers.end())
+		while (first2 != tmarkers.end()) {
+			*reindex2++ = {first2->idx, pos++};
+			smarkers.emplace_back(*first2++).enabled = false;
+		}
+	else
+		while (first1 != smarkers.end()) {
+			*reindex1++ = {first1->idx, pos++};
+			tmarkers.emplace_back(*first1++).enabled = false;
+		}
+
+	const auto markers_size =
+	    std::max(smarkers.size(), tmarkers.size());
+	for (std::size_t ix{}; ix < markers_size; ++ix) {
+		auto &smarker = smarkers[ix];
+		if (auto &[idx, prePos] =
+		        smarker.prevMainMarker.values[0].value;
+		    idx != ~Marker::MarkerIndex{}) {
+			if (prePos < source_reindex.size()
+			    && source_reindex[prePos].idx == idx)
+				prePos = source_reindex[prePos].pos;
+			else if (prePos < target_reindex.size()
+			         && target_reindex[prePos].idx == idx)
+				prePos = target_reindex[prePos].pos;
+			else
+				throw std::runtime_error(
```

Comment:
internal error in the message

Reply:
https://github.com/vizzuhq/vizzu-lib/pull/555/files#diff-afcdf1b3efa39a42c582a4c0fed90d990e1a5405760c8c04a1450c2f48eb4536R176-R179